### ISSUE TITLE
recognition of chords symbols

### ISSUE THREAD

**Issue by Bacchushlg** [Audiveris/audiveris#243]
It would be fine to be able to assign text as chord symbols. Ideally identical to the behavior of lyrics: allow to span a lasso over the chord symbols and define as "chord symbol".
It is clear that the recognition is a bit different from normal text. Especially it is necessary to recognize the flat-b as "b".
Ideally there should be a set of allowed characters.
A...G, H (for german chords)
b, #, m, M, +, -, o, (o with / inside, for m7b5) (a triangle, for maj7)
sus, aug, add, maj, min, mi, alt

MuseScore allows all these variants of chord definition. So there is no need to interprete the chords.

I open a new task, although #44 already asked for this - but it is overloaded with other themes meanwhile

**Comment by webfischi** [comment]
Additional Symbols that are used are:
dim, dom, add, / (commonly used for 6/9 chords) and a very rare version of this,
 with both numbers written over each other like a fraction but without the fraction bar.
![chord symbols](https://user-images.githubusercontent.com/22318963/51852549-9bbb8d00-2326-11e9-9e05-644651a0ad4b.PNG)

The one with the slash inside and the triangle: ø, Δ
There are superscript variations, with just numbers and with numbers and text

**Comment by Bacchushlg** [comment]
thank you for this table.
Just a minor correction: 
- the triangle stands for maj7
- 7+  means maj7, not 7#5

I have built up a table with the harmony expressions, used in musicxml, and the mostly found representations in scores:

[Akkordsymbole.xlsx](https://github.com/Audiveris/audiveris/files/2807060/Akkordsymbole.xlsx)

It's for sure not complete, but it should be valid for 95% of the symbols that appear in typical scores.

**Comment by hbitteur** [comment]
@Bacchushlg and @webfischi 

The main difficulty is to parse the name of a chord, which is very often composed of letters intermixed with other signs (for example, a small circle to indicate a diminished chord).

The **BIG** challenge is to train Tesseract to recognize these names from the image pixels. 
This is a problem on its own, that would require the help of someone really knowledgeable about Tesseract.

Could you take the action point of finding such person?

**Comment by Bacchushlg** [comment]
> According to that only the upper case triangle is maj7 when it's written normal it is only major, very confusing...

Here I found an interesting hint on Wikipedia:
https://en.wikipedia.org/wiki/Chord_names_and_symbols_(popular_music)#Seventh_chords
(see the notice [a] for major seventh)

> But + for #5 or augmented is correct, at least according to the previous mentioned website, other websites 

It seems that also here there exist 2 possibilities: normally the + sign before the 7 means augmented, while 7+ means maj7
See e.g. here:
https://de.wikipedia.org/wiki/Akkordsymbol#Übersicht (-> maj7)

**Comment by Bacchushlg** [comment]
> The BIG challenge is to train Tesseract to recognize these names from the image pixels.
> This is a problem on its own, that would require the help of someone really knowledgeable about Tesseract.
> 

My idea was to use a different traindata file just for the chord recognition, that only contain those characters that are expected for chords. Here the "special" characters (triangle, circle, striked circle) could be assigned to some (for chords) unused characters that later can be properly mapped for the final analysis.

**Comment by hbitteur** [comment]
@Bacchushlg 
You are right. The OCR environment for chord names would be a specific one, different from the environment for the rest of the sheet.
In Tesseract, such environment is selected via a "language" name, such as "eng", "deu", "ita", and is passed as a parameter when Tesseract is called on an image buffer.

So, we would certainly need to build from scratch such a specific "language", trained on combination of all characters (letters and special signs) that can appear in a chord name, plus perhaps a dictionary (I don't remember if the dictionary is really mandatory).

How could we name such language? 
"omr" could be a candidate, although it does not convey its focus on chord names...
Any other suggestion?

**Comment by hbitteur** [comment]
@webfischi 
Thanks for the link to TesseractStudio.Net
Unfortunately, it requires Tesseract version 4 and, as of this writing, Audiveris can operate only with version 3.04, because of the need for JavaCpp interface. Once Tesseract provides this interface for their version  4, we'll be able to make the move.

**Comment by webfischi** [comment]
> It seems that also here there exist 2 possibilities: normally the + sign before the 7 means augmented, while 7+ means maj7
> See e.g. here:
> [https://de.wikipedia.org/wiki/Akkordsymbol#Übersicht](https://de.wikipedia.org/wiki/Akkordsymbol#%C3%9Cbersicht) (-> maj7)

I think I got it now, C7+5 would be C7#5 (because there is also a +9 and a +11 variant) and C7+ could be Cmaj7 the problem is I don't now any examples and Sibelius doesn't show an option for this chord variant.

> maybe "hmn" for HarMoNy. MuseScore handles chord names as such.

There is similar software for older versions:
http://vietocr.sourceforge.net/training.html
